api/v1alpha3/virtualmachine_types.go

```diff
@@ -420,6 +420,27 @@ type VirtualMachineSpec struct {
 
 	// +optional
 
+	// EncryptionClassName describes the name of the EncryptionClass resource
+	// used to encrypt this VM.
+	//
+	// Please note, this field is not required to encrypt the VM. If the
+	// underlying platform implements the vSphere Native Key Provider (NKP) then
```

Comment:
if a default Key Provider is defined in the underlying platform...

It does not necessarily has to be native key provider, it can be standard key provider but it must be set as default in vCenter.